contracts/gas-estimation/InterchainGasEstimation.sol

```diff
@@ -61,14 +62,15 @@ abstract contract InterchainGasEstimation is IInterchainGasEstimation {
         bytes calldata /* params */
     ) public view returns (uint256 gasEstimate) {
         GasInfo storage gasInfo = _storage().gasPrices[destinationChain];
+        GasEstimationType gasEstimationType = GasEstimationType(gasInfo.gasEstimationType);
 
         gasEstimate = gasInfo.axelarBaseFee + (executionGasLimit * gasInfo.relativeGasPrice);
 
         // if chain is L2, compute L1 data fee using L1 gas price info
-        if (gasInfo.gasEstimationType != GasEstimationType.Default) {
+        if (gasEstimationType != GasEstimationType.Default) {
             GasInfo storage l1GasInfo = _storage().gasPrices['ethereum'];
 
-            gasEstimate += computeL1DataFee(gasInfo.gasEstimationType, payload, l1GasInfo);
+            gasEstimate += computeL1DataFee(gasEstimationType, payload, l1GasInfo, gasInfo.l1FeeScalar);
```

Comment:
Unnecessary read potentially, pass gasInfo as storage variable?

Reply:
It's already read with the other vars from this slot. This is a free storage read at this point.

Reply:
Yeah, although not sure how the compiler optimizes this. I do prefer to pass in the slot if possible for consistency, similar to `l1GasInfo`

Reply:
Yeah looks better this way